### ISSUE TITLE
[[ObjectMapper sharedInstance] dictionaryFromObject:object] doesn't work if the object comes from a framework

### ISSUE THREAD

**Issue by fatuhoku** [aryaxt/OCMapper#36]
I didn't change the code — all I did was move it to another pod.

**Comment by aryaxt** [comment]
what do you mean by doesn't work? What's the behavior?

Would it work if you update the code according to this PR? https://github.com/aryaxt/OCMapper/pull/34/files

**Comment by fatuhoku** [comment]
@aryaxt Hi aryaxt, thanks for the super quick reply.

Yes I noticed the cause is definitely the bundle checking.

```objc
...
	// For example when we are mapping an array of string, we shouldn't try to map the string objects inside the array
	if ([NSBundle mainBundle] != [NSBundle bundleForClass:object.class] && [object class] != [NSArray class])
	{
		return object;
	}
```

I think the pull request might go some way to fixing it... I'll need to try that.

**Comment by fatuhoku** [comment]
@aryaxt I checked out the version from the pull request but it's not helped.

I don't understand why bundle checking is necessary at all.  

I'm using AppCode, not Xcode. When I debugged the comparison between `mainBundlePath` and `classBundlePath`, I found their values were WILDLY different:

```objc
mainBundlePath = {__NSCFString * | 0x7ff013d6eb20} "/Users/myusername/Library/Developer/CoreSimulator/Devices/4CCEBA75-D7CE-408F-8135-97927736A940/data/Containers/Bundle/Application/854F84EA-647C-4B99-A94B-E5E94186714B/MakeEatSeeRNUI_Example.app"
classBundlePath = {__NSCFString * | 0x7ff016005020} "/Users/myusername/Library/Caches/AppCode32/DerivedData/MakeEatSeeRNUI-8ccb4a77/Build/Products/Debug-iphonesimulator/MakeEatSeePresenters.framework"
```

So yeah, that comparison would definitely fail, and cause `-dictionaryFromObject:object` to return `object`. Not very useful at all.

**Comment by aryaxt** [comment]
yeah the code to detect project-specific classes need to be reworked. 
To understand the reason behind that logic, you can pull my code, comment out the code, and run the unit tests. Fell free to open a PR if you find a solution. I'll look into it myself as well

**Comment by fatuhoku** [comment]
Hmmm. The relevant test in question is this:

```objc
- (void)testShouldMapArrayOfStringFromObjectToDictionary
{
	User *user = [[User alloc] init];
	user.randomKeywords = @[@"keyword1", @2].mutableCopy;
	
	NSDictionary *dictionary = [self.mapper dictionaryFromObject:user];
	NSArray *array = [dictionary objectForKey:@"randomKeywords"];
	
	XCTAssertTrue(array.count == 2);
	XCTAssertTrue([array[0] isEqualToString:@"keyword1"]);
	XCTAssertTrue([array[1] isEqualToNumber:@2]);
}
```

:/ the implementation appears to make quite a lot of assumptions about what classes belong in bundles and which don't. I've got a wild idea: how about check for the `NS` prefix from the class name instead?

**Comment by aryaxt** [comment]
Let me think about the NS prefix, maybe it could be in additions to bundle checking. Bundle checking is very reliable as long as your models are in the main bundle

**Comment by fatuhoku** [comment]
@aryaxt Yeah I gave **just** checking for the `NS` prefix a quick try, but one of the tests still failed.

**Comment by aryaxt** [comment]
@fatuhoku Did you manage to find a solution? I haven't had time to look into this

**Comment by fatuhoku** [comment]
@aryaxt Yes — I just used [HRCoder](https://github.com/nicklockwood/HRCoder) + [AutoCoding](https://github.com/nicklockwood/AutoCoding) to serialise the JSON out for React Native consumption instead!  I'm not sure how they solve the cross-bundle issue. Probably worth checking the `HRCoder` code.
